v2/protocol/http/protocol_retry.go

```diff
@@ -34,6 +34,11 @@ func (p *Protocol) do(ctx context.Context, req *http.Request) (binding.Message,
 
 func (p *Protocol) doOnce(req *http.Request) (binding.Message, protocol.Result) {
 	resp, err := p.Client.Do(req)
+
+	if resp != nil {
+		defer resp.Body.Close()
```

Comment:
This happens when the returned `binding.Message` has `.Finish` called here: https://github.com/cloudevents/sdk-go/blob/main/v2/protocol/http/protocol.go#L143-L146

This change will break the contract with the the `binding.Message`; So there is something else going on. Still looking.